game/scripts/vscripts/heroes/hero_yukari/yukari_moon_portal.lua

```diff
@@ -6,150 +6,158 @@ LinkLuaModifier("modifier_yukari_moon_portal_caster", "heroes/hero_yukari/yukari
+    end
 end
 
 --function yukari_moon_portal:GetCustomCastErrorTarget(target)
-	--if target == self:GetCaster() and self:GetCaster():IsRooted() then
-		--return "dota_hud_error_ability_disabled_by_root"
-	--end
+--if target == self:GetCaster() and self:GetCaster():IsRooted() then
+--return "dota_hud_error_ability_disabled_by_root"
+--end
 --end
 
-function yukari_moon_portal:OnSpellStart( params )
-	local caster = self:GetCaster()
-	local target = self:GetCursorTarget()
-	-- Handler on lifted targets
-	if caster:HasModifier("modifier_yukari_moon_portal_caster") then
-		local target_loc = self:GetCursorPosition()
-
-	self.target:SetOrigin( target_loc )
-	FindClearSpaceForUnit( self.target, target_loc, true )
-
-
-	else
-
-		self.target = self:GetCursorTarget()
-
-		local duration
-		local is_ally = true
-		-- Create modifier and check Linken
-		if self.target:GetTeam() ~= caster:GetTeam() then
-			if self.target:TriggerSpellAbsorb(self) then
-				return nil
-			end
-			duration = self:GetSpecialValueFor("enemy_lift_duration")
-  if self.target == self:GetCaster() then
-			self.target:AddNewModifier(caster, self, "modifier_yukari_muted", { duration = duration + 0.5 })
-			self.target:AddNewModifier(caster, self, "modifier_yukari_tp_3", { duration = duration })
-			--EmitSoundOn( "yukari.train", caster )
-			is_ally = false
-	else
-	     self.target:AddNewModifier(caster, self, "modifier_yukari_muted", { duration = duration + 0.5 })
-		self.target:AddNewModifier(caster, self, "modifier_yukari_tp", { duration = duration })
-			--EmitSoundOn( "yukari.car", caster )
-			is_ally = false
-			end
-
-		else
-		 if self.target == self:GetCaster() then
-		duration = self:GetSpecialValueFor("ally_lift_duration")
-		self.target:AddNewModifier(caster, self, "modifier_yukari_tp_3", { duration = duration})
-		--EmitSoundOn( "yukari.car2", caster )
-		else
-		duration = self:GetSpecialValueFor("ally_lift_duration")
-		self.target:AddNewModifier(caster, self, "modifier_yukari_tp", { duration = duration})
-		--EmitSoundOn( "yukari.slash", caster )
-		end
-	end
-
-
-
-
-
-		-- Add the particle & sounds
-
-		caster:EmitSound("yukari.portal")
-
-
-		-- Modifier-Params
-
-		-- Add caster handler
-		caster:AddNewModifier(caster, self, "modifier_yukari_moon_portal_caster", { duration = duration - 0.2})
-		caster:AddNewModifier(caster, self, "modifier_yukari_leashed", { duration = duration + FrameTime()})
-		self:EndCooldown()
-	end
+function yukari_moon_portal:OnSpellStart(params)
+    local caster = self:GetCaster()
+    local target = self:GetCursorTarget()
+    local targetLoc = target:GetOrigin()
+    -- Handler on lifted targets
+    if caster:HasModifier("modifier_yukari_moon_portal_caster") then
+        local teleportLoc = self:GetCursorPosition()
+        -- for caster self
+        local toSelfBuff = caster:FindModifierByNameAndCaster("modifier_yukari_tp_3")
```

Comment:
```suggestion
        local toSelfBuff = caster:FindModifierByNameAndCaster("modifier_yukari_tp_3", caster)
```
这个API需要2个参数，只传一个运行时没有报错吗？
https://moddota.com/api/#!/vscripts/CDOTA_BaseNPC#FindModifierByNameAndCaster

```suggestion
        local toSelfBuff = target:FindModifierByNameAndCaster("modifier_yukari_tp_3", caster)
```
考虑到对别的英雄释放的情况，应该在target目标英雄身上找buff。

Reply:
想想不对，E技能2段 鼠标是对空地释放的，`GetCursorTarget()` 真的能取到目标英雄吗。。